edge-modules/DirectMethodSender/Settings.cs

```diff
@@ -38,16 +43,29 @@ public class Settings
             TimeSpan directMethodDelay,
             Option<Uri> analyzerUrl,
             InvocationSource invocationSource,
-            Option<string> serviceClientConnectionString)
+            Option<string> serviceClientConnectionString,
+            string moduleId,
+            TimeSpan testDuration,
+            TimeSpan testStartDelay,
+            Option<Uri> testResultCoordinatorUrl,
+            Option<string> trackingId)
         {
+            Preconditions.CheckRange(testDuration.Ticks, 0);
+            Preconditions.CheckRange(testStartDelay.Ticks, 0);
+
             this.DeviceId = Preconditions.CheckNonWhiteSpace(deviceId, nameof(deviceId));
             this.TargetModuleId = Preconditions.CheckNonWhiteSpace(targetModuleId, nameof(targetModuleId));
             Preconditions.CheckArgument(TransportType.IsDefined(typeof(TransportType), transportType));
             this.TransportType = transportType;
             this.DirectMethodDelay = directMethodDelay;
+            this.InvocationSource = invocationSource;
             this.AnalyzerUrl = analyzerUrl;
             this.ServiceClientConnectionString = serviceClientConnectionString;
-            this.InvocationSource = invocationSource;
+            this.ModuleId = Preconditions.CheckNonWhiteSpace(moduleId, nameof(moduleId));
+            this.TestDuration = testDuration;
+            this.TestStartDelay = testStartDelay;
+            this.TestResultCoordinatorUrl = testResultCoordinatorUrl;
+            this.TrackingId = trackingId;
```

Comment:
I think you should validate tracking id if testresultcoordinatorurl is provided

Reply:
I wouldn't necessary want to do that in case if a developer wants to deploy the code locally and too lazy to provide a trackingId since it doesn't really provide an additional value for a local smoke test.

Reply:
I agree with Philip that it should be validated, sorry for lazy developers 

---
In reply to: [361519203](https://github.com/Azure/iotedge/pull/2172#discussion_r361519203) [](ancestors = 361519203)